python/common_tool_methods.py

```diff
@@ -0,0 +1,151 @@
+        commits = pr.get_commits()
+        for single_commit in commits:
+            cp_result = run(f"git cherry-pick {single_commit.commit.sha}")
+            print(f"Cherry pick result for PR No {pr.number} and commit sha {single_commit.commit.sha}: {cp_result}  ")
+
+
+def get_version_details(version: is_version(str)) -> Dict[str, str]:
+    version_parts = version.split(".")
+    return {"major": version_parts[0], "minor": version_parts[1], "patch": version_parts[2]}
+
+
+def is_major_release(version: is_version(str)) -> bool:
+    version_info = get_version_details(version)
+    return version_info["patch"] == "0"
+
+
+def str_array_to_str(str_array: List[str]) -> str:
+    result_str = ""
+    for i in range(len(str_array)):
+        result_str = result_str + str_array[i] + "\n"
+    return result_str
+
+
+def get_prs(repo: Repository.Repository, earliest_date: datetime, base_branch: str, last_date: datetime = None):
+    pulls = repo.get_pulls(state="all", base=base_branch, sort="created", direction="desc")
+    picked_pulls = []
+    for pull in pulls:
+        if not pull.is_merged():
+            continue
+        if last_date is not None and pull.created_at > last_date:
+            continue
+        if pull.created_at < earliest_date:
+            break
+        if pull.merged_at is not None and pull.merged_at > earliest_date:
+            picked_pulls.append(pull)
+    return picked_pulls
+
+
+def get_pr_issues_by_label(prs: List[PullRequest.PullRequest], label_name: str):
+    filtered_prs = []
+    for pr in prs:
+        filtered_labels = filter(lambda label: label.name == label_name, pr.labels)
+        if len(list(filtered_labels)) > 0:
+            filtered_prs.append(pr)
+    return filtered_prs
+
+
+def has_file_include_line(base_path: str, relative_file_path: str, line_content: str) -> bool:
+    with open(f"{base_path}/{relative_file_path}", "r") as reader:
+        content = reader.read()
+        lines = content.splitlines()
+        found = False
+        for line in lines:
+            if line == line_content:
+                found = True
+                break
+    return found
+
+
+def line_count_in_file(base_path: str, relative_file_path: str, line_content: str) -> int:
+    with open(f"{base_path}/{relative_file_path}", "r") as reader:
+        content = reader.read()
+        lines = content.splitlines()
+        counter = 0
+        for line in lines:
+            if line == line_content:
+                counter = counter + 1
+
+    return counter
+
+
+def replace_line_in_file(file: str, match_regex: str, replace_str: str) -> bool:
+    with open(file, "r") as reader:
+        file_content = reader.read()
+        lines = file_content.splitlines()
+        line_counter = 0
+        has_match = False
+        for line in lines:
+            if re.match(match_regex, line):
+                has_match = True
+                lines[line_counter] = replace_str
+            line_counter = line_counter + 1
```

Comment:
In general, we can use `enumerate` instead of doing `range(len(iterable))` or keeping an counter

```suggestion
        for line_number, line in enumerate(lines):
            if re.match(match_regex, line):
                has_match = True
                lines[line_number] = replace_str
```

Reply:
Fixed by https://github.com/citusdata/tools/pull/91/commits/a2e8aef98e41c785588ddb62c2ed0f902bee180d